### PR TITLE
Add xADRequestODJ resource - fixes #84

### LINKED ISSUE #84: Add xProvisionODJ Resource
Create a new resource for provisioning a computer in AD for ODJ. The ODJ file created can then be consumed by the xOfflineDomainJoin resource I've just [submitted to xComputerManagement](https://github.com/PowerShell/xComputerManagement/pull/27).

The main purpose of this is to allow automated provisioning of Nano Server via DSC, although it will work for other any machine.

DJOIN will be used to provision the machine in AD (/REUSE is required if already exists) and create the request blob:
```
djoin /PROVISION /DOMAIN CONTOSO.COM /MACHINE NANO1 /SAVEFILE c:\nano1-odj.txt
```

### PR DESCRIPTION
**Pull request by PlagueHO**
This change is [Reviewable](https://reviewable.io/reviews/powershell/xactivedirectory/86)

### CODE REVIEW

**Comment by TravisEz13** [comment]
Reviewed 5 of 7 files at r1.
Review status: 5 of 7 files reviewed at latest revision, 3 unresolved discussions.

---

*[DSCResource.Tests, line 1 \[r1\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KF7j1ueFsTNSOb6tqyJ:-KF7j1ueFsTNSOb6tqyK:1353640211) ([raw file](https://github.com/powershell/xactivedirectory/blob/98c7f7f3835f618264807e95d72c628578bd4f0f/DSCResource.Tests#L1)):*
Is there a way to get rid of this?  Doesn't the new test template just clone it?  Is that adequate?  @karolkaczmarek 

---

*[README.md, line 209 \[r1\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KF7jkDW9EQigvjFqzTn:-KF7jkDW9EQigvjFqzTo:645585954) ([raw file](https://github.com/powershell/xactivedirectory/blob/98c7f7f3835f618264807e95d72c628578bd4f0f/README.md#L209)):*
Can you include a link to ODJ documentation such as:
https://technet.microsoft.com/en-us/library/offline-domain-join-djoin-step-by-step(v=ws.10).aspx

---

*[DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.schema.mof, line 1 \[r1\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KF7k6AZghurt6adq9by:-KF7k6AZghurt6adq9bz:1087397062) ([raw file](https://github.com/powershell/xactivedirectory/blob/98c7f7f3835f618264807e95d72c628578bd4f0f/DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.schema.mof#L1)):*
Request should not be part of the name.
I'm not sure I like the ODJ friendly name.  @hemantmahawar , what do you think?
AD implies domain, so I think this might work:
xADOfflineJoinFile

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xactivedirectory/86)*

File: DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.psm1 [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+    $Parameters = @{
+        searchbase = $Domain.DistinguishedName
+        filter = { name -eq $ComputerName }
+    }
+
+    $Computer = Get-ADComputer @Parameters
+    if ($Computer)
+    {
+        $returnValue += @{
+            OU = ($Computer.DistinguishedName -replace "CN=$ComputerName,",'')
```
Comment (iainbrighton) [inline DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.psm1]: You could use the Get-ADObjectParentDN method in the MSFT_xADCommon module.

File: DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.psm1 [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+        [System.String]
+        $DomainName,
+
+        [parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $ComputerName,
+
+        [System.String]
+        $OU,
```
Comment (iainbrighton) [inline DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.psm1]: For consistency with the other resources, should we name this to `Path`?

File: DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.psm1 [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+
+        [System.String]
+        $DCName,
```
Comment (iainbrighton) [inline DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.psm1]: For consistency with the other resources, should we name this to `DomainController`? What about credentials (assuming WMF 4)?

**Comment by PlagueHO** [comment]
Excellent points all round. Thanks gents.

---

Review status: 1 of 7 files reviewed at latest revision, 7 unresolved discussions.

---

*[DSCResource.Tests, line 1 \[r1\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KF7j1ueFsTNSOb6tqyJ:-KF8j5xM0NEtjt6VzMlS:-1954650261) ([raw file](https://github.com/powershell/xactivedirectory/blob/98c7f7f3835f618264807e95d72c628578bd4f0f/DSCResource.Tests#L1)):*
The latest test templates does clone this automatically. I agree - I think we should indeed get rid of this. @IainBrighton? 

---

*[README.md, line 209 \[r1\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KF7jkDW9EQigvjFqzTn:-KF8jfhpw5tE02qkIgkQ:-1044898837) ([raw file](https://github.com/powershell/xactivedirectory/blob/98c7f7f3835f618264807e95d72c628578bd4f0f/README.md#L209)):*
Very good! Done.

---

*[DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.psm1, line 65 \[r1\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KF7LEm9so2eYTvNR_M2-r1-65:-KF8jrtWwCqtcY5ejz4q:492817807) ([raw file](https://github.com/powershell/xactivedirectory/blob/98c7f7f3835f618264807e95d72c628578bd4f0f/DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.psm1#L65)):*
Excellent- I have done this.

---

*[DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.psm1, line 90 \[r1\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KF7LEm9so2eYTvNR_M2-r1-90:-KF8lEMKSBM_FiVNJWEi:1488682622) ([raw file](https://github.com/powershell/xactivedirectory/blob/98c7f7f3835f618264807e95d72c628578bd4f0f/DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.psm1#L90)):*
Good call. Done.

---

*[DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.psm1, line 93 \[r1\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KF7LEm9so2eYTvNR_M2-r1-93:-KF8lGqeB3QUsm8grEEl:1296911082) ([raw file](https://github.com/powershell/xactivedirectory/blob/98c7f7f3835f618264807e95d72c628578bd4f0f/DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.psm1#L93)):*
Good call - I've changed this to DomainController. Although I'm not sure about the credentials. The DJOIN command unfortunately doesn't have any credential parameters. Could use Start-Process to run it under different creds. Your thoughts?

---

*[DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.schema.mof, line 1 \[r1\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KF7k6AZghurt6adq9by:-KF8mwF8ZbyEVb6lnqB7:1427287673) ([raw file](https://github.com/powershell/xactivedirectory/blob/98c7f7f3835f618264807e95d72c628578bd4f0f/DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.schema.mof#L1)):*
Sure! Easy to change - I'm happy with xADOfflineJoinFile. I'll wait to hear from @hemantmahawar before making the change.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-:-KF8mPq4Kq9GgYoOP8Cu:-522739741)*

**Comment by TravisEz13** [comment]
Reviewed 5 of 6 files at r2.
Review status: 6 of 7 files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xactivedirectory/86)*

File: DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.psm1 (hunk 2 above)
Comment (iainbrighton) [inline DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.psm1]: DOH! Looks like you've renamed the output file parameter to $Path, not the $OU one. The xADUser and xADOrganizationalUnit (and the impending xADComputer) use $Path for the target DN.

**Comment by PlagueHO** [comment]
Review status: 6 of 7 files reviewed at latest revision, 4 unresolved discussions.

---

*[DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.psm1, line 90 \[r1\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KF7LEm9so2eYTvNR_M2-r1-90:-KFBsVvEXAY4tuMFJ1oa:-813220146) ([raw file](https://github.com/powershell/xactivedirectory/blob/98c7f7f3835f618264807e95d72c628578bd4f0f/DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.psm1#L90)):*
Ahaha - LOL - it was a late night change. Silly me! I shall correct this pronto :)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xactivedirectory/86)*

**Comment by PlagueHO** [comment]
Review status: 2 of 7 files reviewed at latest revision, 4 unresolved discussions.

---

*[DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.psm1, line 93 \[r1\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KF7LEm9so2eYTvNR_M2-r1-93:-KFBwAhFTDRFmsvIJsqz:929983285) ([raw file](https://github.com/powershell/xactivedirectory/blob/98c7f7f3835f618264807e95d72c628578bd4f0f/DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.psm1#L93)):*
Although, using Start-Process will mean detecting an error from DJOIN is a bit more difficult.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xactivedirectory/86)*

**Comment by TravisEz13** [comment]
Reviewed 1 of 6 files at r2, 5 of 5 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.schema.mof, line 1 \[r1\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KF7k6AZghurt6adq9by:-KFB5-aLVvawdR4eLho0:420858023) ([raw file](https://github.com/powershell/xactivedirectory/blob/98c7f7f3835f618264807e95d72c628578bd4f0f/DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.schema.mof#L1)):*
Discussed with @hemantmahawar and we came up with xADDomainJoinFile.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xactivedirectory/86)*

**Comment by PlagueHO** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.schema.mof, line 1 \[r1\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KF7k6AZghurt6adq9by:-KFCCNfCvjlUK1oADFcD:-1593048331) ([raw file](https://github.com/powershell/xactivedirectory/blob/98c7f7f3835f618264807e95d72c628578bd4f0f/DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.schema.mof#L1)):*
Cool. Thanks @TravisEz13 . I'll get to it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xactivedirectory/86)*

**Comment by PlagueHO** [comment]
Review status: 1 of 7 files reviewed at latest revision, 3 unresolved discussions.

---

*[DSCResources/MSFT_xADDomainJoinFile/MSFT_xADDomainJoinFile.schema.mof, line 1 \[r1\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KF7k6AZghurt6adq9by:-KFCFqS2UlsZD3LAT3So:-499146009) ([raw file](https://github.com/powershell/xactivedirectory/blob/98c7f7f3835f618264807e95d72c628578bd4f0f/DSCResources/MSFT_xADDomainJoinFile/MSFT_xADDomainJoinFile.schema.mof#L1)):*
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xactivedirectory/86)*

**Comment by PlagueHO** [comment]
Review status: 1 of 7 files reviewed at latest revision, 3 unresolved discussions.

---

*[DSCResources/MSFT_xADDomainJoinFile/MSFT_xADDomainJoinFile.psm1, line 90 \[r1\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KF7LEm9so2eYTvNR_M2-r1-90:-KFCFy0pjqDUkUmzm6NG:-499146009) ([raw file](https://github.com/powershell/xactivedirectory/blob/98c7f7f3835f618264807e95d72c628578bd4f0f/DSCResources/MSFT_xADDomainJoinFile/MSFT_xADDomainJoinFile.psm1#L90)):*
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xactivedirectory/86)*

**Comment by TravisEz13** [comment]
@iainbrighton Can you go into reviewable and acknowledge the discussions, otherwise, I'll have to dismiss them before merging.

---

Reviewed 11 of 11 files at r4.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[DSCResources/MSFT_xADDomainJoinFile/MSFT_xADDomainJoinFile.schema.mof, line 1 \[r1\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KF7k6AZghurt6adq9by:-KFHGx1-JO6EFUgmthOq:464619452) ([raw file](https://github.com/powershell/xactivedirectory/blob/98c7f7f3835f618264807e95d72c628578bd4f0f/DSCResources/MSFT_xADDomainJoinFile/MSFT_xADDomainJoinFile.schema.mof#L1)):*
LGTM, I've asked the AD team for comments ( when I proposed the name.)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-:-KFHHBXqiGs5lbNl_PQl:-1652742928)*

**Comment by iainbrighton** [comment]
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[DSCResource.Tests, line 1 \[r1\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KF7j1ueFsTNSOb6tqyJ:-KFBHdDbPEXrgtD0Wmgw:-470029732) ([raw file](https://github.com/powershell/xactivedirectory/blob/98c7f7f3835f618264807e95d72c628578bd4f0f/DSCResource.Tests#L1)):*
I've added an entry into the .gitignore in my xADComputer repo to exclude this in future, but I've not pushed this yet. If you want to push this change with this resource, feel free!

---

*[DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.psm1, line 93 \[r1\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KF7LEm9so2eYTvNR_M2-r1-93:-KFBH4tKGaYOffhqHgCv:-848713844) ([raw file](https://github.com/powershell/xactivedirectory/blob/98c7f7f3835f618264807e95d72c628578bd4f0f/DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.psm1#L93)):*
I wouldn't worry about it if the .exe doesn't support it. If running on a DC this shouldn't be needed. WMF 5 provides PsDscRunAsCredential.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xactivedirectory/86)*

**Comment by iainbrighton** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[README.md, line 1012 \[r4\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KFInzGCvoyEpZKo-AUi:-KFInzGCvoyEpZKo-AUj:-922857212) ([raw file](https://github.com/powershell/xactivedirectory/blob/56c4609079fb679e5e747fd8047beb4c3eb456a6/README.md#L1012)):*
"Domiain" typo

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xactivedirectory/86)*

**Comment by PlagueHO** [comment]
Review status: 6 of 7 files reviewed at latest revision, 3 unresolved discussions.

---

*[DSCResource.Tests, line 1 \[r1\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KF7j1ueFsTNSOb6tqyJ:-KFIraerJO5dqO2V-87-:-289199543) ([raw file](https://github.com/powershell/xactivedirectory/blob/98c7f7f3835f618264807e95d72c628578bd4f0f/DSCResource.Tests#L1)):*
If you've already done it in your PR I'll leave it to yours - I don't want to cause a merge conflict on your PR in case I do it differently! :+1:

---

*[README.md, line 1012 \[r4\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KFInzGCvoyEpZKo-AUi:-KFIrJTFMg0jyH90PI4b:-953483486) ([raw file](https://github.com/powershell/xactivedirectory/blob/56c4609079fb679e5e747fd8047beb4c3eb456a6/README.md#L1012)):*
Doh! Thanks. Fixed

---

*[DSCResources/MSFT_xADDomainJoinFile/MSFT_xADDomainJoinFile.psm1, line 93 \[r1\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KF7LEm9so2eYTvNR_M2-r1-93:-KFIrMdkS48VoFroSmAf:1265636336) ([raw file](https://github.com/powershell/xactivedirectory/blob/98c7f7f3835f618264807e95d72c628578bd4f0f/DSCResources/MSFT_xADDomainJoinFile/MSFT_xADDomainJoinFile.psm1#L93)):*
Ok. Will do. Cheers.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xactivedirectory/86)*

**Comment by iainbrighton** [comment]
Reviewed 1 of 10 files at r2, 10 of 11 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xactivedirectory/86)*

**Comment by TravisEz13** [comment]
Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[DSCResources/MSFT_xADDomainJoinFile/MSFT_xADDomainJoinFile.schema.mof, line 1 \[r1\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KF7k6AZghurt6adq9by:-KFLuk-0DNC_8f-qIDfN:-1418322210) ([raw file](https://github.com/powershell/xactivedirectory/blob/98c7f7f3835f618264807e95d72c628578bd4f0f/DSCResources/MSFT_xADDomainJoinFile/MSFT_xADDomainJoinFile.schema.mof#L1)):*
Sorry for this, but after review with the AD team, the resource should probably be xADComputer.
RequestFile should be optional.  For now, you can throw and say not implemented, if they don't specify request file, but we shouldn't have two resources which creates computer accounts.

What do you think?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xactivedirectory/86)*

**Comment by PlagueHO** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[DSCResources/MSFT_xADDomainJoinFile/MSFT_xADDomainJoinFile.schema.mof, line 1 \[r1\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KF7k6AZghurt6adq9by:-KFLvNvfG4q86PJZ6wBR:957904910) ([raw file](https://github.com/powershell/xactivedirectory/blob/98c7f7f3835f618264807e95d72c628578bd4f0f/DSCResources/MSFT_xADDomainJoinFile/MSFT_xADDomainJoinFile.schema.mof#L1)):*
Won't this conflict with @IainBrighton 's xADComputer resource that he's creating (for creating normal AD Computer objects)?

We could possibly combine the functionality of Offline Domain Joins with @IainBrighton 's new xADCOmputer resource, by calling DJOIN before a Computer object is created IF the RequestFile parameter is set, and then using ```Set-ADComputer``` to configure any further properties on the computer object. If RequestFile parameter is not set then the behavior of xADComputer would remain the same. @IainBrighton - what do you think? You've seen my ODJ resource code, do you think it's possible to combine the functions?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xactivedirectory/86)*

**Comment by TravisEz13** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[DSCResources/MSFT_xADDomainJoinFile/MSFT_xADDomainJoinFile.schema.mof, line 1 \[r1\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KF7k6AZghurt6adq9by:-KFM7c_WSIfc1ffcgPKL:1835942705) ([raw file](https://github.com/powershell/xactivedirectory/blob/98c7f7f3835f618264807e95d72c628578bd4f0f/DSCResources/MSFT_xADDomainJoinFile/MSFT_xADDomainJoinFile.schema.mof#L1)):*
@plagueho  @iainbrighton  Yeah, That is the general idea.  What do you think?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xactivedirectory/86)*

**Comment by iainbrighton** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[DSCResources/MSFT_xADDomainJoinFile/MSFT_xADDomainJoinFile.schema.mof, line 1 \[r1\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KF7k6AZghurt6adq9by:-KFcdYgbOR8pXWYLzmMt:-458351480) ([raw file](https://github.com/powershell/xactivedirectory/blob/98c7f7f3835f618264807e95d72c628578bd4f0f/DSCResources/MSFT_xADDomainJoinFile/MSFT_xADDomainJoinFile.schema.mof#L1)):*
Of course it's possible. Given enough time and effort!

I have pushed my current xADComputer implementation (still need to finish the unit tests) to https://github.com/iainbrighton/xactivedirectory/tree/Issue69. At least we can look at merging into a single code base.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xactivedirectory/86)*

**Comment by PlagueHO** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[DSCResources/MSFT_xADDomainJoinFile/MSFT_xADDomainJoinFile.schema.mof, line 1 \[r1\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/86#-KF7k6AZghurt6adq9by:-KFcly8pUgojcUFyi1N3:1724443360) ([raw file](https://github.com/powershell/xactivedirectory/blob/98c7f7f3835f618264807e95d72c628578bd4f0f/DSCResources/MSFT_xADDomainJoinFile/MSFT_xADDomainJoinFile.schema.mof#L1)):*
Cool! I'll put a hold on this resource until your one is committed. I'll then see if I can merge what I've done into your xADComputer resource. Might not be till next week as this week is a bit busy.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xactivedirectory/86)*

**Comment by PlagueHO** [comment]
I'm closing this PR and will resubmit a new PR with the ODJ stuff added to the new xADComputer resource.

**Pull request closed without merging**